### PR TITLE
Avoid zero scale in StudentTOutput

### PR DESCRIPTION
**Pull request by shchur**
*Issue #, if available:* #2790

*Description of changes:*
- Avoid zero scale in `StudentTOutput `. Similar to the fix in #1893.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by kashif** [comment]
I think we should also move from softplus to squareplus but yes this is a helpful fix!

File: src/gluonts/torch/distributions/studentT.py [hunk 1]
```diff
@@ -67,7 +68,8 @@ class StudentTOutput(DistributionOutput):
     def domain_map(
         cls, df: torch.Tensor, loc: torch.Tensor, scale: torch.Tensor
     ):
-        scale = F.softplus(scale)
+        epsilon = np.finfo(cls._dtype).eps  # machine epsilon
```
Comment (kashif) [inline src/gluonts/torch/distributions/studentT.py]:
```suggestion
        epsilon = torch.finfo(scale.dtype).eps
```

Comment (shchur) [reply src/gluonts/torch/distributions/studentT.py]: Nice, thank you!

File: src/gluonts/torch/distributions/studentT.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 
 from typing import Dict, Tuple, Union
 
+import numpy as np
```
Comment (jaheba) [inline src/gluonts/torch/distributions/studentT.py]: Looks unused.

**Comment by jaheba** [comment]
Is there a way we can test this?

File: src/gluonts/torch/distributions/studentT.py (hunk 2 above)
Comment (shchur) [reply src/gluonts/torch/distributions/studentT.py]: Thanks for catching this

**Comment by shchur** [comment]
@jaheba The logic here looks very similar to other `DistributionOutput` classes (e.g., [Beta](https://github.com/awslabs/gluonts/blob/dev/src/gluonts/torch/distributions/distribution_output.py#L193) and [Gamma](https://github.com/awslabs/gluonts/blob/dev/src/gluonts/torch/distributions/distribution_output.py#L213)). If there are existing tests that cover these, we could extend them to StudentT.

The only thing that comes to my mind would be training a network with respective model types on data consisting of all zeros and seeing if it eventually runs into NaNs. However, even this is not guaranteed and may take a a non-deterministic number of epochs.

File: src/gluonts/torch/distributions/studentT.py (hunk 2 above)
Comment (jaheba) [reply src/gluonts/torch/distributions/studentT.py]: `ruff` did, I was just pointing it out!

**Review by jaheba — APPROVED** [review]

**Comment by jaheba** [comment]
I see, I guess similar to the distribution tests we have for mxnet, the test would add little value, is and just slows down the test suite.

**Pull request merged**